src/pages/[...slug].tsx

```diff
@@ -101,13 +101,16 @@ export default function Page({ page }: PageProps): JSX.Element {
         <>
             <NextSeo title={page.frontMatter?.title || page.title} description={page.frontMatter?.description} />
             <div className="container">
-                <nav id="right-sidebar">
-                    <section className="right-sidebar-section" ref={tocReference}>
-                        <h4 className="sidebar-heading">On this page</h4>
-                        <TableOfContents toc={page.toc} className="table-of-contents" />
-                    </section>
-                    <EditSection page={page} />
-                </nav>
+                {!page.frontMatter?.hide_sidebar ? (
```

Comment:
tip: Instead of a ternary with `null`, you can just use `&&` (which is the conventional way to do it in JSX)